src/modules/map/components/LocationPopupCopySlot.vue

```diff
@@ -1,102 +1,121 @@
-    </button>
-</template>
-
-<script>
+<script setup>
 import tippy from 'tippy.js'
+import { computed, onBeforeUnmount, onMounted, ref, watch } from 'vue'
 import { useI18n } from 'vue-i18n'
-import { mapState } from 'vuex'
+import { useStore } from 'vuex'
 
 import log from '@/utils/logging'
 
-export default {
-    props: {
-        value: {
-            type: String,
-            default: '',
-        },
+const { identifier, value, extraValue, resetDelay } = defineProps({
+    identifier: {
+        type: String,
+        required: true,
     },
-    setup() {
-        const i18n = useI18n()
-        return {
-            i18n,
-        }
+    value: {
+        type: String,
+        required: true,
     },
-    computed: {
-        ...mapState({
-            lang: (state) => state.i18n.lang,
-        }),
+    extraValue: {
+        type: String,
+        default: null,
     },
-    watch: {
-        lang() {
-            this.setTooltipContent()
-        },
+    resetDelay: {
+        type: Number,
+        default: 1000,
     },
-    mounted() {
-        this.copyTooltip = tippy('#copyButton', {
-            arrow: true,
-            placement: 'right',
-            touch: 'hold',
-        })
+})
 
-        this.copiedTooltip = tippy('#copyButton', {
-            arrow: true,
-            placement: 'right',
-            trigger: 'click',
-            onShow(instance) {
-                setTimeout(() => {
-                    instance.hide()
-                }, 1000)
-            },
-            allowHTML: true,
-        })
-        this.setTooltipContent()
-    },
-    unmounted() {
-        this.copyTooltip?.forEach((tooltip) => tooltip.destroy())
-        this.copiedTooltip?.forEach((tooltip) => tooltip.destroy())
-    },
-    methods: {
-        async copyValue() {
-            try {
-                await navigator.clipboard.writeText(this.value)
-            } catch (error) {
-                log.error(`Failed to copy to clipboard:`, error)
-            }
-        },
-        setTooltipContent() {
-            this.copyTooltip?.forEach((instance) => {
-                instance.setContent(this.i18n.t('copy_cta'))
-            })
-            this.copiedTooltip?.forEach((instance) => {
-                instance.setContent(this.i18n.t('copy_done'))
-            })
-        },
-    },
+const copyButton = ref(null)
+const copied = ref(false)
+
+const i18n = useI18n()
+
+const store = useStore()
+const lang = computed(() => store.state.i18n.lang)
+
+const buttonIcon = computed(() => {
+    if (copied.value) {
+        return 'check'
+    }
+    // as copy is part of the "Regular" icon set, we have to give the 'far' identifier
+    return ['far', 'copy']
+})
+
+let copyTooltip = null
+
+onMounted(() => {
+    copyTooltip = tippy(copyButton.value, {
+        arrow: true,
+        placement: 'right',
+        hideOnClick: false,
+        // no tooltip on mobile/touch
+        touch: false,
+    })
```

Comment:
The `allowHTML: true,` is required due the `&nbsp;` in the french translation of `copy_done`